libraries/botbuilder-dialogs/src/memory/scopes/settingsMemoryScope.ts

```diff
@@ -25,16 +35,110 @@ export class SettingsMemoryScope extends MemoryScope {
      * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) object for this turn.
      * @returns The memory for the scope.
      */
-    public getMemory(dc: DialogContext): object {
-        let settings: object = {};
+    public getMemory(dc: DialogContext): Record<string, unknown> {
+        let settings: Record<string, unknown> = {};
         if (dc.context.turnState.has(ScopePath.settings)) {
             settings = dc.context.turnState.get(ScopePath.settings);
-        }
-        for (const key in process.env) {
-            if (typeof process.env[key] == 'string') {
-                settings[key] = process.env[key];
+        } else {
+            const configuration = dc.context.turnState.get(DialogTurnStateConstants.configuration);
+            if (configuration) {
+                settings = SettingsMemoryScope.loadSettings(configuration);
+                dc.context.turnState.set(ScopePath.settings, settings);
```

Comment:
The configuration value in the TurnState may come from multiple sources (e.g. `process.env` or `TestScript.configuration`). 

In our initial version of `SettingsMemoryScope`, `process.env` is the only source to load settings so I added a key `ScopePath.settings` to allow loading settings from TurnState to load complex settings (`precess.env` are only key/value pairs) via https://github.com/microsoft/botbuilder-js/pull/2217 . The settings from `process.env` were kept in order to keep backward compatibility.

I would prefer merging configuration from TurnState with `process.env` and load them into settings by calling `loadSettings(mergedConfiguration)`.

Reply:
It's unfortunate that this code is in `botbuilder-dialogs` otherwise I would have said to not worry about backwards compatibility... Lessons for the future. Thanks for the explanation @chon219!